src/modules/simulation/gz_bridge/GZBridge.cpp

```diff
@@ -122,15 +122,44 @@ int GZBridge::init()
 		bool result;
 		std::string create_service = "/world/" + _world_name + "/create";
 
-		if (_node.Request(create_service, req, 1000, rep, result)) {
-			if (!rep.data() || !result) {
-				PX4_ERR("EntityFactory service call failed");
-				return PX4_ERROR;
+		bool gz_called = false;
+		// Check if STANDALONE has been set.
+		char *standalone_val = std::getenv("STANDALONE");
+
+		if ((standalone_val != nullptr) && (std::strcmp(standalone_val, "1") == 0)) {
+			// Check if Gazebo has been called and if not attempt to reconnect.
+			while (gz_called == false) {
+				if (_node.Request(create_service, req, 1000, rep, result)) {
+					if (!rep.data() || !result) {
+						PX4_ERR("EntityFactory service call failed");
+						return PX4_ERROR;
+
+					} else {
+						gz_called = true;
+					}
+				}
+
+				// If Gazebo has not been called, wait 2 seconds and try again.
+				else {
+					printf("WARN  [gz_bridge] Service call timed out as Gazebo has not been detected. \n");
```

Comment:
```suggestion
					PX4_WARN("Service call timed out as Gazebo has not been detected.");
```

Reply:
@dagar There is a bug where if you try to write coloured text to a file using PX4_WARN it will not appear. Using printf, this does not happen.

Reply:
Testing this now

Reply:
Ok. Doesn't seem to be an issue any longer.